numpyro/infer/elbo.py

```diff
@@ -994,25 +1001,35 @@ def single_particle_elbo(rng_key):
                         )
                         for key in deps
                     ]
-
-                if dice_factors:
-                    dice_factor = reduce(lambda a, b: a + b, dice_factors)
-                    cost = cost * funsor.ops.exp(dice_factor)
-                if (scale is not None) and (not is_identically_one(scale)):
-                    cost = cost * to_funsor(scale)
```

Comment:
IIUC those computations are performed in eager contract tensor?

Reply:
Which line are you referring to? 
Just to be clear eager contract tensor is used in cases that look like this:
``` 
Compute expectation of cost(b) wrt q(a)q(b|a):
# eliminate `a` first:
logq(b) = eager_contract_tensor(Variable(a), [logq(a), logq(b|a)])
# then compute expectation wrt q(b):
cost = sum(cost(b) * logq(b).exp())
```